File: Codechef folder/Level 1 to 2/twodish.py
#TWODISH
t=int(input())
for i in range(t):
    n,a,b,c=map(int,input().split())
    if b>=n:
        if a+c>=n:
            print("YES")
        else:
            print("NO")
    else:
        print("NO")

File: Codechef folder/Level 1 to 2/twodish.cpp
//TWODISH
#include<iostream>
using namespace std;
int main()
{
    int t,n,a,b,c;
    cin>>t;
    while(t--)
    {
        cin>>n>>a>>b>>c;
        if(b>=n)
        {
            a+c>=n?cout<<"YES"<<endl:cout<<"NO"<<endl;
        }
        else
            cout<<"NO"<<endl;
    }
}